arcaflow_plugin_rtla/rtla_plugin.py

```diff
@@ -78,9 +68,46 @@ def run_timerlat(
                 f"{err.cmd[0]} failed with return code {err.returncode}:\n{err.output}"
             )
 
+        if params.enable_time_series:
+            timeseries_dict = {}
+            trace_path = "/sys/kernel/debug/tracing/instances/timerlat_hist/trace_pipe"
+
+            timeseries_cmd = ["cat", trace_path]
+
+            # A delay is needed before reading from the trace_path to ensure the file
+            # exists and data is streaming to the pipe. I've tried to avoid a sleep(),
+            # but none of the methods I've tested have worked.
+            wait_time = 0
+            timeout_seconds = 5
+            sleep_time = 0.5
+            trace_file_exists = False
+            while not trace_file_exists and wait_time < timeout_seconds:
+                time.sleep(sleep_time)
+                trace_file_exists = os.path.isfile(trace_path)
+                wait_time += sleep_time
+
+            if not trace_file_exists:
+                print(
+                    "Unable to read tracer output; " "Skipping time series collection\n"
+                )
+            else:
+                try:
+                    timeseries_proc = subprocess.Popen(
+                        timeseries_cmd,
+                        start_new_session=True,
+                        stdout=subprocess.PIPE,
+                        stderr=subprocess.PIPE,
+                        text=True,
+                    )
+                except subprocess.CalledProcessError as err:
+                    return "error", ErrorOutput(
+                        f"""{err.cmd[0]} failed with return code {err.returncode}:\n
+                        {err.output}"""
+                    )
+
         try:
             # Block here, waiting on the cancel signal
-            print("Gathering data... Use Ctrl-C to stop.")
+            print("Gathering data... Use Ctrl-C to stop.\n")
```

Comment:
This looks odd now. At this point, your timeseries latency info is fully gathered... you ran the `cat`, and it will quickly return whatever is in the trace file so far. If the intent is to gather data while you're running RTLA, something like Webb's `tail -F` would work better, as it'll continue pulling new lines until you stop it. (Or defer launching your "one-and-done" `cat` until after you get the stop signal.)

Reply:
I suspect that `cat` exits when it reaches the "end of file but that the file in question doesn't have an end (i.e., I think it's a pipe); so at this point, the parent process is waiting for a duration while the _two_ subprocesses are both actively running.  So, `cat` would work fine except that it's pickier about the lifetime of the file than `tail -F` is.

Reply:
The file being read here is a fifo, and (interestingly) `tail -f` does not work on it.  Using `cat` here keeps the fifo read open until we later explicitly terminate the subprocess. Note that this `subprocess` has now been changed to take the fifo file as stdin instead of as a parameter of the `cat` command.

Reply:
> `tail -f` does not work on it.

Did you try `tail -F` (note the uppercase `F`)?

(This is presumably moot now that you've added the additional logic to wait for the FIFO to appear, etc.)